test/e2e/api_inheritance/api_inheritance_test.go

```diff
@@ -88,9 +80,6 @@ func TestAPIInheritance(t *testing.T) {
 				ctx = withDeadline
 			}
 
-			require.Equal(t, 1, len(f.Servers), "incorrect number of servers")
-			server := f.Servers[serverName]
```

Comment:
:+1:

All these should go. Not sure why they were added in the first place. Lack of trust into the fixtures?

Reply:
unrelated to this PR: can we have a helper for the context logic above? It's painful to see that everywhere.

Reply:
> unrelated to this PR: can we have a helper for the context logic above? It's painful to see that everywhere.

I'm happy to strip the context logic out entirely. With `t.Cleanup` it should be possible for anyone who wants to use a context to instantiate local to use vs relying on anything passed around.